src/plugins/auto_batch/src/sync_infer_request.cpp

```diff
@@ -49,30 +50,32 @@ size_t SyncInferRequest::get_batch_size() const {
     return m_batch_size;
 }
 
-void SyncInferRequest::share_tensors_with_batched_req(const std::set<std::string>& batched_inputs,
-                                                      const std::set<std::string>& batched_outputs) {
+void SyncInferRequest::share_tensors_with_batched_req(const std::set<std::size_t>& batched_inputs,
+                                                      const std::set<std::size_t>& batched_outputs) {
     for (const auto& it : get_inputs()) {
-        auto name = ov::op::util::get_ie_output_name(it);
+        size_t port_hash = ov::util::hash_combine(
+            std::vector<size_t>{std::hash<const ov::Node*>()(it.get_node()), std::hash<size_t>()(it.get_index())});
```

Comment:
why do  we need port hash? I supposed we can use index of this port among all input ports. 

E.g. replace for-range loop with for-index loop and use port index

Reply:
I suppose we need node and its index to determine one port, so hash is used to combine them as one size_t data. It will impact performance, maybe we can figure out better solution.

Reply:
Since inputs (aka parameters) order is always fixed, it should be enough to use port index only as @ilya-lavrenov suggested

Reply:
> why do we need port hash? I supposed we can use index of this port among all input ports.
> 
> E.g. replace for-range loop with for-index loop and use port index

@ilya-lavrenov If we have multi-inputs/outputs nodes & the node has more than one port, the index will be the same I think.

Reply:
Input node is always Parameter, it has only one output port.